share/functions/funced.fish

```diff
@@ -83,10 +83,16 @@ function funced --description 'Edit function definition'
     or return 1
     set -l tmpname $tmpdir/$funcname.fish
 
-    if functions -q -- $funcname
-        functions -- $funcname >$tmpname
-    else
+    set -l writepath
+
+    if not functions -q -- $funcname
         echo $init >$tmpname
+    else if functions --details -- $funcname | string match --invert --regex '^-$'
```

Comment:
Okay, so this path is for functions that have been defined in a file?

Note that interactively defined functions say "stdin" here, the `-` bit only applies to functions `source`d from stdin.

We might want to clean that up - see src/builtin_functions.cpp.

Reply:
Thanks, I'll add that to the expression.